src/box/iproto.cc

```diff
@@ -2789,6 +2857,12 @@ net_end_join(struct cmsg *m)
 	iproto_msg_delete(msg);
 
 	assert(! ev_is_active(&con->input));
+	con->is_in_replication = false;
+
+	if (iproto_shutdown_started) {
```

Comment:
This global variable is updated from the tx thread so accessing it from the iproto thread without synchronization looks unsafe. Also, I think we should maintain this flag per connection so that we can drop existing connections without shutting down iproto for the sake of https://github.com/tarantool/tarantool-ee/issues/585.

Reply:
Replaced with per connection flag.